web/src/dialog_widget.ts

```diff
@@ -232,7 +215,8 @@ export function launch(conf: DialogWidgetConfig): string {
             return;
         }
         if (conf.loading_spinner) {
-            show_dialog_spinner();
+            const $button = $("#dialog_widget_modal .modal__btn");
+            loading.show_modal_spinner($button);
```

Comment:
I think we probably want to preserve the `show_dialog_spinner` abstraction; the updated code here is a lot less readable. because of the need to look up the button 

But `show_dialog_spinner` could call `loading.show_modal_spinner` internally?

Reply:
No,right now I have replaced the show_dialog_spinner with show_modal_spinner function.
And talking about preserving the abstraction of `show_dialog_spinner`,I will make sure to maintain this by cleaning up my code.